### PR TITLE
#2200 - Add Redesign PostTabCard Component

### PR DESCRIPTION
**Pull request by linadinh2602**
### What does this pull request aim to achieve? 💯
image
- PostTabCard take in an array of cardContent number of maxPost, look like this: 
          <PostTabCard cardContents={[
            {
              title: 'Request',
              posts: Object.values(posts.posts),
            },
            {
              title: 'Offer',
              posts: Object.values(posts.posts),
            }
          ]} />
- if maxPost is not given, it'll default to 3
- Currently, the menu dropdown doesn't have any callback implemented yet based on the discussion because it'll be changing in the future.
- This PR doesn't integrate any component yet
_Please be concise and link any related issue(s):_ #2200 

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** for a regular feature, **`hotfix/<branch_name>`** for a hotfix, and **`service/<branch_name>`** for a microservice, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Review by clubkemp — COMMENTED** [review]
What exactly are you modifying in the Post.js file? It looks like a lot of formatting changes, did you actually make changes to that component that effect PostTabCard? If you did could you let me know some line numbers so I can review, otherwise should discard the changes??

**Comment by linadinh2602** [comment]
@clubkemp  I didn't change the Post.js, I didn't realized it was updated due to the linter on my VSCode. You can safely ignore the change in that file.

File: client/src/components/Feed/Post.js [hunk 1]
```diff
@@ -467,17 +463,17 @@ const Post = ({
...
         </div>
       }
       thumb={
         post?.author?.photo ? (
           post.author.photo
         ) : (
-          <TextAvatar>{AvatarName}</TextAvatar>
-        )
+            <TextAvatar>{AvatarName}</TextAvatar>
+          )
```
Comment (linadinh2602) [inline client/src/components/Feed/Post.js]: Will revert changes in Post.js. This is updated by accident.

**Review by clubkemp — APPROVED** [review]

**Review by aditilamkhede — CHANGES_REQUESTED** [review]
@linadinh2602 it looks good, just few things.
1. As we want this component to be reusable, I would suggest to move all styled components to separate js. and export them. That way any body can reuse them. 
2. wherever px are used, if you can use rem as unit, please do so.
3. Also put your formatdate function in util so it becomes reusable if needed.
4. Add a comment over your functions, what it does.

File: client/src/components/Feed/PostTabCard.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  )
+}
+
+const PostTabCard = ({ cardContents, maxPosts = 3 }) => {
+  const tabs = cardContents.map((item) => ({ title: item.title }))
+  return (
+    <Container>
+      <Tabs tabs={tabs}
+        initialPage={1}
+        onChange={(tab, index) => { console.log('onChange', index, tab); }}
```
Comment (aditilamkhede) [inline client/src/components/Feed/PostTabCard.js]: remove these logs too

**Comment by linadinh2602** [comment]
@aditilamkhede Sure I can expose everything in a separate file, but I'm worried that making everything public (reusable) by default will make things more confusing. quoted this article: https://itnext.io/thinking-in-styled-components-e230ea37c52c
"In short, don’t default to making things reusable just for the heck of it, because there are tradeoffs. It’s not necessarily better to make things as reusable as possible."

**Comment by aditilamkhede** [comment]
> @aditilamkhede Sure I can expose everything in a separate file, but I'm worried that making everything public (reusable) by default will make things more confusing. quoted this article: https://itnext.io/thinking-in-styled-components-e230ea37c52c
> "In short, don’t default to making things reusable just for the heck of it, because there are tradeoffs. It’s not necessarily better to make things as reusable as possible."

right @linadinh2602 but in our case our main aim was to create reusable components, like your post card, we have to create Request and Offer cards too. those styles which you created for post card, we are going to create many cards with same style. I hope you get my point here. Also for your formatdate function, you may also need it fir request and offer cards.

**Comment by aditilamkhede** [comment]
> > @aditilamkhede Sure I can expose everything in a separate file, but I'm worried that making everything public (reusable) by default will make things more confusing. quoted this article: https://itnext.io/thinking-in-styled-components-e230ea37c52c
> > "In short, don’t default to making things reusable just for the heck of it, because there are tradeoffs. It’s not necessarily better to make things as reusable as possible."
> 
> right @linadinh2602 but in our case our main aim was to create reusable components, like your post card, we have to create Request and Offer cards too. those styles which you created for post card, we are going to create many cards with same style. I hope you get my point here. Also for your formatdate function, you may also need it fir request and offer cards.

Also I may have wrote by mistake all styles from postcard.js but the ones which will be reusable just expose those.

**Review by aditilamkhede — APPROVED** [review]

**Pull request merged**